src/main/java/ru/spbstu/service/impl/VariantServiceImpl.java

```diff
@@ -43,4 +47,19 @@ public void saveWithBatch(final InputStream inputStream,
                 .map(vcfRecord -> Variant.newInstance(vcfRecord, dbName));
         Iterators.partition(variants.iterator(), saveBatchSize).forEachRemaining(variantRepository::saveAll);
     }
+
+    @Override
+    public Optional<Variant> find(String chromosome, Long position, String referenceBase, String alternateBase) {
+        return variantRepository.findVariant(chromosome, position, referenceBase, alternateBase);
+    }
+
+    @Override
+    public VariantListDto getAnnotatedVariants(VariantListDto requestedVariantListDto) {
+        List<Variant> annotatedVariants = requestedVariantListDto.getVariants().stream()
+                .map(var -> find(var.getChromosome(), var.getPosition(), var.getReferenceBase(), var.getAlternateBase()))
```

Comment:
Not to fix now, but just a friendly recommendation for future coding: try some stream and optional excercises. It's useful and fun! :)

Reply:
Yeah good exercise can be to rewrite this without calling `Optional.get` and `Optional.isPresent`.
Please avoid using these methods when possible (in some rare cases they can be used).
Consider using `Optional.stream()` method to convert optional to stream.

Reply:
Thank you! Just starting get familiar with Streams so I will study at my leisure :)